Unity-Technologies/ScriptableRenderPipeline
# Custom Function Node and Subgraph Output Node refactor

### Purpose of this PR
- Adds Custom Function Node
- Refactors Subgraph Output Node to support all port types and renaming
- Adds ReorderableSlotList View to handle the above two changes
- Changes Convert To Subgraph to create Properties/out ports with names based on types for Blackboard consistency

**NOTE:** This PR contains an issue that will be fixed by a separate PR (@spaztronaut )
- Deleting a port that has attached edges creates a nullref and error state is not resolved. This reproduces on both Custom Function Node and Subgraph Output. This also reproduces in master with current Subgraph Output node.

---
### Release Notes
- See changelog

---
### Testing status
**Katana Tests**: Running

**Manual Tests**: What did you do?
- Used Custom Function Node a lot with various settings
- Converted various graphs to subgraphs
- Test to fail:
   - Slot/Prop name duplicate sanity
   - Tried to break preview shader with Custom Function node in invalid state
   - Tried to create Convert to Subgraph issues

**Automated Tests**:  Nothing, should be added to graphics tests (in PR)

Any test projects to go with this to help reviewers? Nope

**NOTE**: Make sure to test upgrading from previously serialized subgraphs and graphs (with subgraphs in). @spaztronaut 

---
### Overall Product Risks
**Technical Risk**: Low

**Halo Effect**: Low

---
### Comments to reviewers
N/A
